packages/vm/lib/state/cache.ts

```diff
@@ -51,30 +51,39 @@ export default class Cache {
     }
   }
 
+  /**
+   * Returns true if the key was deleted and thus existed in the cache earlier
+   * @param key - trie key to lookup
+   */
+  keyIsDeleted(key: Buffer): boolean {
+    const keyStr = key.toString('hex')
+    const it = this._cache.find(keyStr)
+    if (it.node) {
+      return it.value.deleted
+    }
+    return false
+  }
+
   /**
    * Looks up address in underlying trie.
    * @param address - Address of account
-   * @param create - Create emtpy account if non-existent
    */
-  async _lookupAccount(address: Buffer, create: boolean = true): Promise<Account | undefined> {
+  async _lookupAccount(address: Buffer): Promise<Account> {
     const raw = await this._trie.get(address)
-    if (raw || create) {
-      const account = new Account(raw)
-      return account
-    }
+    const account = new Account(raw)
+    return account
   }
 
   /**
    * Looks up address in cache, if not found, looks it up
    * in the underlying trie.
    * @param key - Address of account
-   * @param create - Create emtpy account if non-existent
    */
-  async getOrLoad(key: Buffer, create: boolean = true): Promise<Account | undefined> {
```

Comment:
looks good, does this mean the casting i was referencing in this comment can be removed then? https://github.com/ethereumjs/ethereumjs-vm/pull/807#discussion_r454045007

Reply:
Forgot to remove this cast, thanks 😄